### ISSUE TITLE
Evaluate on sets of classes

### ISSUE THREAD

**Issue by adriaromero** [triagemd/keras-eval#55]
Given a model and test set based on N classes, allow the evaluation on sets of classes by providing a *testing dictionary* or similar.

E.g. 
Training scenario:
```
[class_0]
[class_1]
[class_2]
[class_3]
```

Testing scenario:
```
[test_set_0] class_0 or class_1
[test_set_1] class_2 or class_3
```

So the way we combine probabilities is as below:
```
probability(test_set_0) =  probability(class_0) + probability(class_1)
probability(test_set_1) = probability(class_2) + probability(class_3)
```
We would want the users to give us the mapping between the training and testing dictionary as a `.json` file. Given below is the format we expect:
```
[
  {
    "class_index": 0,
    "class_name": "dog",
     "group":"land_animals" 
  },
  {
    "class_index": 1,
    "class_name": "cat",
    "group":"land_animals"
  },
  {
    "class_index": 2,
    "class_name": "gold_fish",
    "group":"sea_creatures"
  }

]
```
So in the example above the group gives us the mapping between a single concept during training and the concepts which we would want to evaluate on in test.
